templates/common/openstack/files/usr-local-bin-openstack-kubelet-nodename.yaml

```diff
@@ -31,15 +31,23 @@ contents:
     # Set node name to be instance name instead of the default FQDN hostname.
     # Try to get the hostname from config drive first if it's available
     # or fallback to the metadata service.
-    if [ -f "/host/var/config/openstack/2012-08-10/meta_data.json" ]; then
+    if [ -z "$(blkid -t LABEL="config-2" -odevice)" ]; then
+        tmp_mount=$(mktemp -d)
+        mount -o ro -t auto /dev/disk/by-label/config-2 $tmp_mount
+        meta_data_file=$tmp_mount/openstack/2012-08-10/meta_data.json
+        if [ ! -f $meta_data_file ]; then
+            echo "Failed to find $meta_data_file"
+            exit 1
+        fi
         # https://docs.openstack.org/nova/victoria/user/metadata.html#metadata-openstack-format
-        hostname=$(jq -re .name /host/var/config/openstack/2012-08-10/meta_data.json)
+        hostname=$(jq -re .name $meta_data_file)
         if [[ -z "${hostname}" ]]; then
-            echo "Failed to get hostname from /host/var/config/openstack/2018-08-10/meta_data.json"
+            echo "Failed to get hostname from $meta_data_file"
             exit 1
         fi
 
         echo "KUBELET_NODE_NAME=${hostname}" > ${NODEENV}
+        umount $tmp_mount
```

Comment:
do we need to unmount? or is it to keep the pattern done in ignition?

Reply:
> do we need to unmount? or is it to keep the pattern done in ignition?

yes, cleaner I guess.


>  Also should it be unmounted inside the first condition?

yes i'll add it.